src/clients/python/examples/simple_grpc_string_infer_client.py

```diff
@@ -108,4 +97,76 @@
             print("error: incorrect difference")
             sys.exit(1)
 
+
+def identity_inference(triton_client, np_array):
+    model_name = "simple_identity"
+    inputs = []
+    outputs = []
+
+    inputs.append(grpcclient.InferInput('INPUT0', np_array.shape, "BYTES"))
+    inputs[0].set_data_from_numpy(np_array)
+
+    outputs.append(
+        grpcclient.InferRequestedOutput('OUTPUT0'))
+
+    results = triton_client.infer(model_name=model_name,
+                                  inputs=inputs,
+                                  outputs=outputs)
+    if (np_array.dtype == np.object_):
+        print(results.as_numpy('OUTPUT0'))
+        if not np.array_equal(np_array, results.as_numpy('OUTPUT0')):
+            print(results.as_numpy('OUTPUT0'))
+            print("error: incorrect output")
+            sys.exit(1)
+    else:
+        encoded_results = np.char.encode(
+            results.as_numpy('OUTPUT0').astype(str))
+        if not np.array_equal(np_array, encoded_results):
+            print(encoded_results)
+            print("error: incorrect output")
+            sys.exit(1)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v',
+                        '--verbose',
+                        action="store_true",
+                        required=False,
+                        default=False,
+                        help='Enable verbose output')
+    parser.add_argument('-u',
+                        '--url',
+                        type=str,
+                        required=False,
+                        default='localhost:8001',
+                        help='Inference server URL. Default is localhost:8001.')
+
+    FLAGS = parser.parse_args()
+    try:
+        triton_client = grpcclient.InferenceServerClient(url=FLAGS.url,
+                                                         verbose=FLAGS.verbose)
+    except Exception as e:
+        print("context creation failed: " + str(e))
+        sys.exit()
+
+    # Example using BYTES input tensors with utf-8 encoded string
+    # elements
+    simple_string_inference(triton_client)
+
+    # Example using BYTES input tensor with utf-8 encoded string that
+    # has an embedded null character.
+    null_chars_array = np.array(
+        ["he\x00llo".encode('utf-8') for i in range(16)], dtype=np.object_)
```

Comment:
Actually, this worked even before the new changes were introduced. It would be great if we can have two arrays here where one of them uses `np.object` and the other one uses `np.bytes_`. Then, we can show in the examples that trailing zeros are lost when using `np.bytes_`.

```python
null_chars_object = np.array(
        ["hello\x00".encode('utf-8') for i in range(16)], dtype=np.object_)
identity_inference(null_chars_object)
null_chars_bytes = np.array(
        ["hello\x00".encode('utf-8') for i in range(16)], dtype=np.bytes_)
incorrect_identity_inference(null_chars_object)
```

Where the `incorrect_identity_inference` passes when results don't match.

Reply:
Can you file a ticket for this addition.

Reply:
https://jirasw.nvidia.com/browse/DLIS-2213